src/hotspot/share/code/nmethod.cpp

```diff
@@ -2123,10 +2123,27 @@ void nmethod::purge(bool unregister_nmethod) {
 
   // completely deallocate this method
   Events::log_nmethod_flush(Thread::current(), "flushing %s nmethod " INTPTR_FORMAT, is_osr_method() ? "osr" : "", p2i(this));
-  log_debug(codecache)("*flushing %s nmethod %3d/" INTPTR_FORMAT ". Live blobs:" UINT32_FORMAT
-                       "/Free CodeCache:%zuKb",
-                       is_osr_method() ? "osr" : "",_compile_id, p2i(this), CodeCache::blob_count(),
-                       CodeCache::unallocated_capacity(CodeCache::get_code_blob_type(this))/1024);
+
+  LogTarget(Debug, codecache) lt;
+  if (lt.is_enabled()) {
+    ResourceMark rm;
+    const char* name = method()->name()->as_C_string();
+    const char* is_jvmci = "";
+    const size_t codecache_capacity = CodeCache::capacity()/1024;
+    const size_t codecache_free_space = CodeCache::unallocated_capacity(CodeCache::get_code_blob_type(this))/1024;
+#if INCLUDE_JVMCI
+    if (is_compiled_by_jvmci()) {
+      if (jvmci_name() != nullptr) {
+        name = jvmci_name();
+      }
+      is_jvmci = "JVMCI compiled ";
+    }
+#endif
+    log_debug(codecache)("Flushing nmethod %3d/" INTPTR_FORMAT ", level=%d, osr=%d, cold=%d, epoch=" UINT64_FORMAT ", cold_count=" UINT64_FORMAT ". "
+                       "Cache capacity: %zuKb, free space: %zuKb. %smethod %s",
+                       _compile_id, p2i(this), _comp_level, is_osr_method(), is_cold(), _gc_epoch, CodeCache::cold_gc_count(),
+                       codecache_capacity, codecache_free_space, is_jvmci, name);
```

Comment:
Swap `is_jvmci` and `name` so that method's names in output for nmethod not compiled by JVMCI (by C1) will be aligned.